pdfarranger/undo.py

```diff
@@ -51,23 +53,36 @@ def commit(self, label):
         :param label: label of the action
         """
         self.states = self.states[:self.current]
-        self.states.append(([row[0].duplicate(False) for row in self.model], self.label,))
+        self.states.append(self.get_state())
         self.current += 1
         self.label = label
         self.__refresh()
 
+    def get_state(self):
+        """
+        Get the content which should be saved:
+
+        1. The label of the action
+        2. The pages
+        3. Which page numbers are selected
+        4. The vertical adjustment percentual value
+        """
+        pages = [row[0].duplicate(False) for row in self.model]
+        s = self.app.iconview.get_selected_items()
+        selection = [path.get_indices()[0] for path in s]
+        vadj_percent = self.app.vadj_percent_handler()
+        return (self.label, pages, selection, vadj_percent)
```

Comment:
Maybe that could be something like a small `@dataclass` instead of a tuple of four values just put together in a tuple. When reviewing this, it really is harder to understand and there is a risk of mixing stuff up and issues when changing/extending those tuples in the future.

(I know we are having code like this in a lot of places, around here. But maybe we could start with cleaner code when adding/modifying stuff.)